crates/driver/src/infra/api/routes/solve/dto/auction.rs

```diff
@@ -42,16 +42,8 @@ impl Auction {
                         },
                         kind: match order.class {
                             Class::Market => competition::order::Kind::Market,
-                            Class::Limit if order.partially_fillable => {
-                                competition::order::Kind::Limit {
-                                    surplus_fee: eth::U256::zero().into(),
-                                }
-                            }
                             Class::Limit => competition::order::Kind::Limit {
-                                surplus_fee: order
-                                    .surplus_fee
-                                    .ok_or(Error::MissingSurplusFee)?
-                                    .into(),
+                                surplus_fee: eth::U256::zero().into(),
```

Comment:
To me this indicates that this field isn't needed at all anymore and can be removed from the type.